src/backend/utils/adt/ri_triggers.c

```diff
@@ -95,29 +96,27 @@
 #define RUN_AS_PSQL(qplan)  																				\
 do																											\
 {																											\
-	bool reset_dialect = (sql_dialect == SQL_DIALECT_TSQL);                                         \
+	bool reset_dialect = (sql_dialect == SQL_DIALECT_TSQL);                                         		\
+	int sql_dialect_old = sql_dialect;																		\
+	void *newextra = NULL;																					\
 	if (reset_dialect)																						\
-		set_config_option("babelfishpg_tsql.sql_dialect", "postgres",										\
-			GUC_CONTEXT_CONFIG,						\
-			PGC_S_SESSION, GUC_ACTION_SAVE, true, 0, false);												\
-																											\
+	{																										\
+		sql_dialect = SQL_DIALECT_PG;																		\
+		assign_sql_dialect(sql_dialect, newextra);															\
+	}																										\
 	PG_TRY();																								\
 	{																										\
 		qplan;																								\
 	}																										\
-	PG_CATCH();																								\
+	PG_FINALLY();																							\
 	{																										\
 		if (reset_dialect)																					\
-			set_config_option("babelfishpg_tsql.sql_dialect", "tsql",										\
-							GUC_CONTEXT_CONFIG,		\
-							PGC_S_SESSION, GUC_ACTION_SAVE, true, 0, false);								\
-		PG_RE_THROW();																						\
```

Comment:
why are we not re-throwing the error?

Reply:
When using `PG_FINALLY()`, explicit calls to `PG_RE_THROW()` are unnecessary. The `PG_END_TRY()` macro automatically handles error re-throwing if an exception occurred during execution.